### PR TITLE
Converting the action into a Typescript Action instead of being a jav…

### PR DESCRIPTION
**Pull request by s-tuli**
Converting the action into a Typescript Action instead of being a javascript action. 
Not using babel anymore. 
For testing: instead of jasmine, using jest.

### CODE REVIEW

File: actions/add-review-url/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import AddComment from './AddReviewUrl';
+import * as core from "@actions/core";
+
+async function run() {
+    const  addComment = new AddComment();
+    try {
+        const host = core.getInput('host');
+        let headref = '';
```
Comment (stepro) [inline actions/add-review-url/src/index.ts]: I assume your goal here is to have headref be typed as `string`, not `string | undefined`. Try this:

```typescript
let headref = process.env.GITHUB_HEAD_REF || '';
```

It's more concise than using an if statement and the toString() (if it's a truthy value, then it's known to be a string).

File: actions/add-review-url/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import AddComment from './AddReviewUrl';
+import * as core from "@actions/core";
+
+async function run() {
+    const  addComment = new AddComment();
```
Comment (stepro) [inline actions/add-review-url/src/index.ts]: Extra space between `const` and `addComment`. If the editor is not able to format this for you, try looking at tslint.

File: actions/add-review-url/package.json [hunk 3]
```diff
@@ -1,29 +1,29 @@
 {
-  "name": "AddReviewUrl",
+  "name": "addcomment",
```
Comment (stepro) [inline actions/add-review-url/package.json]: Why change this?

Comment (s-tuli) [reply actions/add-review-url/package.json]: Updated. (Had created and tested a new Typescript package.json)

File: actions/add-review-url/src/index.ts (hunk 2 above)
Comment (s-tuli) [reply actions/add-review-url/src/index.ts]: Using tslint now. Removed the extra space.

File: actions/add-review-url/src/index.ts (hunk 1 above)
Comment (s-tuli) [reply actions/add-review-url/src/index.ts]: Thanks. Updated it.

**Pull request merged**